README.md

```diff
@@ -539,30 +524,271 @@ services.Configure<TargetingEvaluationOptions>(options =>
 
-To customize the loading of feature definitions, one must implement the `IFeatureDefinitionProvider` interface.
+To customize the loading of feature definitions, one must implement the `IFeatureFlagDefinitionProvider` interface.
 
-```
-public interface IFeatureDefinitionProvider
+``` C#
+public interface IFeatureFlagDefinitionProvider
 {
-    Task<FeatureDefinition> GetFeatureDefinitionAsync(string featureName);
+        Task<FeatureFlagDefinition> GetFeatureFlagDefinitionAsync(string featureName, CancellationToken cancellationToken = default);
 
-    IAsyncEnumerable<FeatureDefinition> GetAllFeatureDefinitionsAsync();
+        IAsyncEnumerable<FeatureFlagDefinition> GetAllFeatureFlagDefinitionsAsync(CancellationToken cancellationToken = default);
 }
 ```
 
 To use an implementation of `IFeatureDefinitionProvider` it must be added into the service collection before adding feature management. The following example adds an implementation of `IFeatureDefinitionProvider` named `InMemoryFeatureDefinitionProvider`.
 
-```
+``` C#
 services.AddSingleton<IFeatureDefinitionProvider, InMemoryFeatureDefinitionProvider>()
         .AddFeatureManagement()
 ```
 
+It is also possible to provide custom dynamic feature definitions. This is done by implementing the `IDynamicFeatureDefinitionProvider` interface and registering it as mentioned above.
+
+## Dynamic Features
+
+When new features are being added to an application there may come a time when a feature has multiple different proposed design options. The different options for the design of a feature can be referred to as variants of the feature, and the feature itself can be referred to as a dynamic feature. A dynamic feature is a feature that can have different values (variants) extending beyond a simple on/off flag. A common pattern when rolling out dynamic features is to surface the different variants of a feature to different segments of a user base and to see how each variant is perceived. The most well received variant could be the one that gets rolled out to the entire user base, or if necessary the feature could be scrapped. There could be other reasons to expose different variants of a feature, for example using a different version every day of the week. The goal of this method is to establish a model that can help solve these common patterns that occur when rolling out features that can have different variants.

```

Comment:
What you described is basically A/B Experiment. Should we refer to the term so it's easy to understand for people who already know the concept?

Reply:
Are you suggesting to scrap most of the explanation, or just add A/B experiment somewhere in the mix?

Reply:
Your content is good. Just mention A/B experiment as an example of a use case.